src/main/java/com/salesforce/mirus/KafkaMonitor.java

```diff
@@ -166,8 +166,9 @@ private String applyRoutersToTopic(String topic) {
     String destBootstrap = config.getDestinationBootstrapServers();
     Map<String, Object> destConsumerProps = config.getDestinationConsumerProperties();
 
-    destConsumerProps.computeIfAbsent(
-        "bootstrap.servers", s -> destConsumerProps.put("bootstrap.servers", destBootstrap));
+    if (!destConsumerProps.containsKey("bootstrap.servers")) {
+      destConsumerProps.put("bootstrap.servers", destBootstrap);
+    }
```

Comment:
`computeIfAbsent` approach would look like
```
destConsumerProps.computeIfAbsent("bootstrap.servers", $ -> destBootstrap);
```
The map uses the given function to produce the value and enters it for you. No need to call `put` within that function.

Reply:
Does that help with the concurrency issue? It feels like this this is just a syntactic difference?

Reply:
The short answer is that `putIfAbsent` is even more appropriate, because the cost of computing the value is not high enough to be worth deferring. But about `computeIfAbsent`:

Yes, it will help with the concurrency exception. The difference is more than syntactic.

In the original version, the map is calling your function, which modifies the map directly. Note also that the return value of the function (the return value of `put`) is the previous value associated with that key (presumably `null` in this case), and per the contract of `computeIfAbsent`, the fact that the function returns `null` means that `computeIfAbsent` would [immediately remove the mapping](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#putIfAbsent-K-V-). You may think that the above doesn't make a lot of sense, or is subject to timing issues caused by the particular implementation of the map, and that's part of the reason why this approach causes a concurrent modification exception.

Reply:
Agreed, `putIfAbsent()` is the better alternative. And yes, the `computeIfAbsent()` seems to depend on the execution context: the unit tests are not failing however the integration tests when a Mirus connector is instantiated hits this error.